login/utils.py

```diff
@@ -0,0 +1,52 @@
+from pathlib import Path
+from typing import Optional
+from urllib.parse import quote
+
+from django.http import HttpRequest, HttpResponse
+from django.conf import settings
+from ipware import get_client_ip
+from netaddr import EUI
+
+class MacAddr:
+    @classmethod
+    def deserialize_from(cls, request: HttpRequest) -> Optional[EUI]:
+        """Pulls mac-address data from the metadata of the request.
+                EUI is used to ensure Mac-address is formatted correctly"""
+        if request.session.get('macaddr') is not None:
+            return EUI(request.session['macaddr'])
+        else:
+            return None
```

Comment:
This is an instane of boolean laundering.

```suggestion
        return None
```